src/model/priors/single_cell_bulk.rs

```diff
@@ -0,0 +1,578 @@
+    fn prob_rho(
+        &self,
+        af_single_underlying: &f64,
+        n_obs: &usize,
+        k: &usize
+    ) -> LogProb
+    {
+        let binomial_coeff = ln_binomial(*n_obs as u64,*k as u64);
+
+        match *af_single_underlying {
+            // model for hom ref sites
+            0.0 => {
+                let alpha = |cov| {
+                    -0.000027183 * cov as f64 + 0.068567471
+                };
+                let beta = |cov| {
+                    0.007454388 * cov as f64 + 2.367486659
+                };
+                let a = alpha(*n_obs);
+                let b = beta(*n_obs);
+
+                LogProb(
+                    binomial_coeff +
+                        ln_beta((*k as f64) + a,  (*n_obs) as f64 - (*k as f64) + b) - ln_beta(a,b)
+                )
+            },
+            // model for heterozygous sites
+            0.5 => {
+                let weight = |cov| {
+                    0.000548761 * cov as f64 + 0.540396786
+                };
+                let alpha_1 = |cov| {
+                    0.057378844 * cov as f64 + 0.669733191
+                };
+                let alpha_2 = |cov| {
+                    0.003233912 * cov as f64 + 0.399261625
+                };
+
+                let a1 = alpha_1(*n_obs);
+                let a2 = alpha_2(*n_obs);
+                let w = weight(*n_obs);
+
+                // Lodato et al. 2015, Science, Supplementary Information, pages 8f and Fig. S5 (A, C, E)
+                LogProb(
+                    w.ln() + binomial_coeff +
+                        ln_beta((*k as f64) + a1,  (*n_obs) as f64 - (*k as f64) + a1) - ln_beta(a1,a1)
+                ).ln_add_exp(
+                    LogProb(
+                        ((-w).ln_1p()) + binomial_coeff +
+                            ln_beta((*k as f64) + a2, (*n_obs) as f64 - (*k as f64) + a2) - ln_beta(a2,a2) )
+                )
+            },
+            // model for hom alt sites (hom ref density mirrored)
+            1.0 => {
+                let alpha = |cov| {
+                    0.007454388 * cov as f64 + 2.367486659
+                };
+                let beta = |cov| {
+                    -0.000027183 * cov as f64 + 0.068567471
+                };
+                let a = alpha(*n_obs);
+                let b = beta(*n_obs);
+
+                LogProb(
+                    binomial_coeff +
+                        ln_beta((*k as f64) + a,  (*n_obs) as f64 - (*k as f64) + b) - ln_beta(a,b)
+                )
+            },
+            _ => panic!("SingleCellBulkModel is currently only implemented for the diploid case with allele frequencies 0.0, 0.5 and 1.0.")
+        }
+    }
+}
+
+
+impl PairModel<DiscreteAlleleFreqs, ContinuousAlleleFreqs> for SingleCellBulkModel {
+
+    fn prior_prob(&self, _: AlleleFreq, _: AlleleFreq, _: Variant) -> LogProb {
+        // TODO: stick in the InfiniteSitesNeutralVariationModel here?
+        LogProb::ln_one()
```

Comment:
Well, in effect, the probabilities for the single cell sample and the bulk sample are currently independent (as are the maximum a posteriori allele frequency estimates), so theoretically a `pub trait Model` for just the single cell sample might do for variant calling in the single cell sample. The bulk so far is only needed for somatic vs. germline classifications in prosolo.

But eventually, the bulk could also provide some kind of prior information for the single cell sample, thus linking the two again.

But as for prob_rho and being a prior: It is prior knowledge that we have, but not on the plain probability of an underlying allele frequency. Rather, it is a full-blown probability mass function of seeing a certain amount `k` of alternative alleles out of the local coverage `n` for a possible underlying allele frequency. A prior would instead just weigh between the possible underlying allele frequencies, right?

For now, I'll add the `panic`.